apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V59__Alter_beneficiaries_bene_id_numeric.sql

```diff
@@ -0,0 +1,9 @@
+--
+-- The goal here is to create a column, bene_id_numeric, on the
+-- beneficiaries table; an ALTER TABLE will lock the table from
+-- table reads (SELECT) which would definitely affect BFD services
+-- but this is simply a meta change and will complete in ms.
+--
+-- Also, it should be noted that unless explicitly disabled, each flyway
+-- script executes within a transaction.
+ALTER TABLE public.beneficiaries ADD bene_id_numeric bigint;
```

Comment:
at this point can we just make it one script?

Reply:
doubtful; even though the alter table runs in a transaction, the alter table by definition creates an exclusive lock cuz it basically has to add the column to every row in the table; it's somewhat analogous to 'vacuum full'. Here is my little cheat-sheet that I use:

Reply:
I've been trying to operate under the assumption, that at most a table would be locked from a read for < 1sec

Reply:
but if you move this inside of your ex-lock in the other script then it already has the lock

Reply:
It would, except I shouldn't have an exclusive lock in the other script; it needs to be:
LOCK TABLE ONLY public.beneficiaries IN SHARE UPDATE EXCLUSIVE MODE;
which allows SELECT (reads) to still happen.

ALTER TABLE is a killer with respect to having minimal impact on reads; by definition, it creates an EXCLUSIVE LOCK, which prevents reads from happening until the lock is released. Testing in the clone last yesterday afternoon confirmed this.

Now I can merge everything into one flyway migration, but it complicates the script a bit; I'll have to have two seperate transaction, each with its own pessimistic locks. I'll run a test now.

Reply:
ah, ok, thanks. looks like the alter is actually getting an 'ACCESS EXCLUSIVE' lock and not just an 'EXCLUSIVE' lock (which would allow reads). https://www.postgresql.org/docs/9.1/explicit-locking.html

I think I am fine with the strategy that you have now